### PR TITLE
Improve reviewability of WebNN operations baseline

### PR DESCRIPTION
**Pull request by dontcallmedom**

### CODE REVIEW

File: src/batch_normalization.js [hunk 1]
```diff
@@ -1,27 +1,9 @@
...
-      throw new Error(`The parameter ${name} is not a 1-D tensor.`);
-    } else {
-      if (a.shape[0] !== expectedSize) {
-        throw new Error(`The length ${a.shape[0]} of the ${name} values is not equal to the ` +
-          `size ${expectedSize} of the input dimension denoted by options.axis.`);
-      }
-    }
-  }
-}
+import {reshape} from './lib/reshape.js';
```
Comment (huningxin) [inline src/batch_normalization.js]: `reshape` is a WebNN op https://www.w3.org/TR/webnn/#api-mlgraphbuilder-reshape

File: src/gru.js [hunk 2]
```diff
@@ -2,12 +2,13 @@
 
 import {concat} from './concat.js';
 import {gruCell} from './gru_cell.js';
-import {reshape} from './reshape.js';
-import {sizeOfShape, Tensor} from './tensor.js';
-import {sigmoid} from './sigmoid.js';
+import {reshape} from './lib/reshape.js';
+import {sizeOfShape, Tensor} from './lib/tensor.js';
+import {sigmoid} from './lib/sigmoid.js';
```
Comment (huningxin) [inline src/gru.js]: `sigmoid` is a WebNN op: https://www.w3.org/TR/webnn/#api-mlgraphbuilder-sigmoid

File: test/batch_normalization_test.js [hunk 3]
```diff
@@ -2,10 +2,10 @@
 
 import {batchNormalization} from '../src/batch_normalization.js';
 import {clamp} from '../src/clamp.js';
-import {leakyRelu} from '../src/leaky_relu.js';
+import {leakyRelu} from '../src/lib/leaky_relu.js';
```
Comment (huningxin) [inline test/batch_normalization_test.js]: `leakyRelu` is a WebNN op: https://www.w3.org/TR/webnn/#api-mlgraphbuilder-leakyrelu

File: src/relu.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-import {unary} from './unary.js';
+import {unary} from './lib/unary.js';
```
Comment (huningxin) [inline src/relu.js]: the ops in unary are WebNN ops: https://www.w3.org/TR/webnn/#api-mlgraphbuilder-unary

File: src/lib/validate-input.js [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+    } else {
+      if (a.shape[0] !== expectedSize) {
+        throw new Error(`The length ${a.shape[0]} of the ${name} values is not equal to the ` +
+          `size ${expectedSize} of the input dimension denoted by options.axis.`);
+      }
+    }
+  }
+}
+
+export function validateInput(op, args) {
```
Comment (huningxin) [inline src/lib/validate-input.js]: Should we have specific method (or even separate file) for each op's validation? Say, `validateConv2dInput` etc.,

**Review by huningxin — COMMENTED** [review]
Thanks much @dontcallmedom ! 

Lint tool reports a few errors. Please take a look. You can run the lint tool locally by `npm run lint`.  If you found lint rules are inappropriate, feel free to update the rules accordingly.

File: src/lib/validate-input.js (hunk 5 above)
Comment (dontcallmedom) [reply src/lib/validate-input.js]: if we keep input validation, and if we keep it separate, separate methods are probably indeed much cleaner; I'm not sure separate files are necessary warranted, but I don't mind strongly either

Comment (huningxin) [reply src/lib/validate-input.js]: Regarding to [discussion](https://github.com/webmachinelearning/webnn-baseline/pull/1#issuecomment-1031184302), it looks like we agree to keep the input validation code. So could you please make separate methods for them? I agree with you we can skip the separate files.

**Review by huningxin — COMMENTED** [review]
Thanks @dontcallmedom for the update. My only open is about separate methods for input validation. Other code LGTM.

**Review by huningxin — APPROVED** [review]
LGTM, great work, thanks much @dontcallmedom !

**Pull request merged**